Intake Position Check

droid-101 opened the pull request:
Intake Subsystem will now check if the intake is retracted or deployed.
- If intake is retracted, motor will not spin to avoid damage to hardware.
- Added key mappings for intake functions in Constants

Commit:
Retracted Intake Check

Intake subsystem will check if the intake is extended and will accordingly allow the intake motor to spin or not.

Commit:
Joystick Bindings

Added constants to hold the values of button mappings on joysticks.

bricktrooper commented on src/main/java/frc/robot/subsystems/Intake.java:
```diff
@@ -12,25 +12,34 @@
   {
     intakeMotor = new CANSparkMax(Constants.INTAKE_CAN_SPARK, MotorType.kBrushless); // Motor is Brushless
     intakePiston = new DoubleSolenoid(Constants.INTAKE_PISTON_DEPLOY, Constants.INTAKE_PISTON_RETRACT);
+
```
Why the random space here?  This is a relatively short function so you don't need extra spaces.  It doesn't make it any more readable.

bricktrooper commented on src/main/java/frc/robot/subsystems/Intake.java:
```diff
@@ -12,25 +12,34 @@
 
   public void deploy()
   {
+    deployed = true;
```
Makes more sense to change the value after you write to the solenoid

bricktrooper commented on src/main/java/frc/robot/subsystems/Intake.java:
```diff
@@ -12,25 +12,34 @@
 
   public void retract()
   {
+    deployed = false;
```
Same comment as above.

bricktrooper commented on src/main/java/frc/robot/subsystems/Intake.java:
```diff
@@ -12,25 +12,34 @@
   public void spin()
   {
-    intakeMotor.set(0.25);
+    if (deployed == true)
```
You don't really need to write `== true` here.  It is already a `boolean` and the compiler will optimize this code out anyway.  It is much easier to read `if (deployed) { do stuff }`.

bricktrooper commented on src/main/java/frc/robot/subsystems/Intake.java:
```diff
@@ -40,7 +49,10 @@ public void stop()
   public void reverse()
   {
-    intakeMotor.set(-0.25);
+    if (deployed == true)
```
Same comment as above

Commit:
Address Review Comments

- Removed unecessary spaces
- Changes value of position tracker after writing to solenoids
